doc/compatibility.rst

```diff
@@ -0,0 +1,105 @@
+Minimum Supported GMT/Python/NumPy Versions
```

Comment:
I'm debating if we should rename the file to `minversions.md`.

Reply:
Done in https://github.com/GenericMappingTools/pygmt/pull/2862/commits/c9c1dd72e286531bd6dfab1f7a0a73c6cb142dcd.